source/ncengine/graphics2/diligent/resource/base/StructuredBuffer.h

```diff
@@ -0,0 +1,126 @@
+#pragma once
+#include "graphics2/ShaderTypes.h"
+#include "ncutility/NcError.h"
+
+#include "Common/interface/RefCntAutoPtr.hpp"
+#include "Graphics/GraphicsEngine/interface/RenderDevice.h"
+#include "Graphics/GraphicsEngine/interface/DeviceContext.h"
+
+#include <concepts>
+#include <span>
+
+namespace nc::graphics
+{
+class StructuredBufferBase
+{
+    public:
+        explicit StructuredBufferBase(std::string_view name,
+                                      Diligent::IShaderResourceVariable& variable,
+                                      uint32_t maxSize,
```

Comment:
change to `maxElementCount` and `initialElementCount` to be consistent with below